### PR TITLE
Throw graphql errors in rhaiscript

### PR DESCRIPTION
**Pull request by o0Ignition0o**
### Throw graphql errors in rhaiscript

Up until now rhai script throws would yield an http status code and a message String which would end up as a GraphQL error.
This change allows users to throw with a valid GraphQL response body, which may include data, as well as errors and extensions.

Refer to the `Terminating client requests` section of the [Rhai api documentation](https://www.apollographql.com/docs/router/configuration/rhai) to learn how to throw GraphQL payloads.

### CODE REVIEW

**Comment by garypen** [comment]
If part of our motivation is to improve compatibility with existing bits, maybe we should expose some of these via the new global variables mechanism?
https://www.apollographql.com/docs/apollo-server/data/errors/#built-in-error-codes

**Comment by o0Ignition0o** [comment]
That's a great idea!

File: apollo-router/src/plugins/rhai/mod.rs [hunk 1]
```diff
@@ -1093,79 +1110,72 @@ impl ServiceStep {
...
+        Self {
+            line: value.line(),
+            pos: value.position(),
         }
     }
 }
 
+#[derive(Deserialize, Debug)]
+struct ErrorDetails {
+    #[serde(
```
Comment (Geal) [inline apollo-router/src/plugins/rhai/mod.rs:1145]: do we really need to bring in http_serde just to serialize the status code?

**Review by Geal — APPROVED** [review]

File: apollo-router/src/plugins/rhai/mod.rs (hunk 1 above)
Comment (o0Ignition0o) [reply apollo-router/src/plugins/rhai/mod.rs:1146]: I don't have any other way in my mind, except implementing it manually :/

**Review by garypen — APPROVED** [review]

**Pull request merged**